public/sample.csv

```diff
@@ -1,4 +1,5 @@
-token_address,receiver,amount
-0x6810e776880c02933d47db1b9fc05908e5386b96,0x1000000000000000000000000000000000000000,0.0001
-0x6b175474e89094c44da98b954eedeac495271d0f,0x2000000000000000000000000000000000000000,0.0001
-,0x3000000000000000000000000000000000000000,0.0001
\ No newline at end of file
+token_type,token_address,receiver,value,id
+erc20,0x6810e776880c02933d47db1b9fc05908e5386b96,0x1000000000000000000000000000000000000000,0.0001
+erc20,0x6b175474e89094c44da98b954eedeac495271d0f,0x2000000000000000000000000000000000000000,0.0001
+native,,0x3000000000000000000000000000000000000000,0.0001
+erc721,0x57f1887a8bf19b14fc0df6fd9b2acc9af147ea85,0x4000000000000000000000000000000000000000,,42
```

Comment:
Could we call the type NFT instead of exactly which implementation? I would imagine some users don't know which type their NFTs actually are.

Reply:
I thought about it and I think I just found the solution for this.
I could use the eip165 standard and use the supportsInterface Method. This has to be present for valid ERC1155 contracts.

> Smart contracts implementing the ERC-1155 standard MUST implement the ERC-165 supportsInterface function and MUST return the constant value true if 0xd9b67a26 is passed through the interfaceID argument.

Also ERC721 includes the ERC-165 standard:
Every ERC-721 compliant contract must implement the ERC721 and ERC165 interfaces. (interfaceID: 0x80ac58cd)

Because I have to be able to detect which kind of contract we have due to the different signatures. For ERC1155 without a given value I could also assume its "1".

Reply:
Fair enough. In that case maybe we can include both an erc721 type and erc1155 type in the sample file.

Reply:
Here is a gif of the new version:
![Uploading new_version.gif…]()

Reply:
Beauty!